### PR TITLE
Shm cmake

### LINKED ISSUE #29: CMake build system

### PR DESCRIPTION
**Pull request by J-Cieplinski**
Implemented CMake build functionality

### CODE REVIEW

File: shm/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.10)
+set(PROJECT_NAME SHM)
```
Comment (lisie-sprawy) [inline shm/CMakeLists.txt]: You can use `project(SHM)` and use (${PROJECT_NAME} as variable in these places when is needed.

File: shm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+project(${PROJECT_NAME})
```
Comment (lisie-sprawy) [inline shm/CMakeLists.txt]: if you correct above, you won't need it

File: shm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+FILE(GLOB SRC_FILES "*.cpp")
```
Comment (lisie-sprawy) [inline shm/CMakeLists.txt:3]: It is probably correct, but in the task we had to create a list of cpp files in the variable

Comment (lisie-sprawy) [reply shm/CMakeLists.txt:3]: we can define cpp files by set

File: shm/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+list(REMOVE_ITEM SRC_FILES main.cpp test.cpp)
+
+
+add_library(${PROJECT_NAME}-lib STATIC ${SRC_FILES})
```
Comment (lisie-sprawy) [inline shm/CMakeLists.txt:14]: I don't know that SCR_FILES has main.cpp or not? Once you define SCR_FILES by file, later by list, so in this situation what happens? :)

**Review by lisie-sprawy — CHANGES_REQUESTED** [review]

File: shm/CMakeLists.txt (hunk 3 above)
Comment (J-Cieplinski) [reply shm/CMakeLists.txt:3]: It does create list of cpp files in variable SRC_FILES. I just don't have to list them one by one and any new ones will be added automatically

File: shm/CMakeLists.txt (hunk 4 above)
Comment (J-Cieplinski) [reply shm/CMakeLists.txt:14]: SRC_FILES created by `files` adds all *.cpp files by default. So I have to remove them from SRC_FILES using this command. Still better than adding stuff one by one :D

File: shm/CMakeLists.txt (hunk 1 above)
Comment (J-Cieplinski) [reply shm/CMakeLists.txt]: done

File: shm/CMakeLists.txt (hunk 4 above)
Comment (lisie-sprawy) [reply shm/CMakeLists.txt:14]: nice, good to know ;)

**Review by lisie-sprawy — APPROVED** [review]

**Review by ArturGubala — APPROVED** [review]

**Pull request merged**